front/src/App.vue

```diff
@@ -67,15 +89,39 @@ export default {
 
       //解析ボタンを画像uploadするまで隠す
       //setFile()でtrueへ
-      isVisible : false
+      isVisible : false,
+      
+      // heif,heicからjpegへの変換中に表示する
+      isConverting: false
     }
   },
   methods: {
-    setFile(e) {
+    async setFile(e) {
       const file = e.target.files[0]
-      this.imgFileInput = file
-      this.imgPreviewUrl = URL.createObjectURL(file)
-      this.isVisible = true
+
+      if (file) {
+        if (file.type === 'image/heif' || file.type === 'image/heic') {
+          // heif,heicの場合プレビュー表示できないのでjpegに変換する
+
+          // 変換に時間がかかるのでモーダルを表示
+          this.isConverting = true
+          const convertedJpeg = await heic2any({
+            blob: file,
+            toType: 'image/jpeg'
+          });
+          // Fileオブジェクトに変換
+          this.imgFileInput = new File([convertedJpeg], 'converted.jpeg', { type: 'image/jpeg' })
+          this.imgPreviewUrl = URL.createObjectURL(convertedJpeg);
+          this.isConverting = false
```

Comment:
```suggestion
        if (file.type === 'image/heif' ||
          file.type === 'image/heic' ||
          file.name.toLowerCase().endsWith('.heif') ||
          file.name.toLowerCase().endsWith('.heic')) {
          // heif,heicの場合プレビュー表示できないのでpngに変換する
          // 変換に時間がかかるのでモーダルを表示
          this.isConverting = true
          const convertedImage = await heic2any({
            blob: file,
            toType: 'image/png'
          });
          this.imgFileInput = convertedImage;
          this.imgPreviewUrl = URL.createObjectURL(convertedImage);
          this.isConverting = false
```

- heicはtypeが取得できないことがある(手元のwindowsでは無理だった)ので拡張子を確認する
- Blobのままで良い、Fileに変換する必要はない

Reply:
- jpegは不可逆変換されるのでpngにする